packages/core/template/project/DeveloperGuide.md

```diff
@@ -0,0 +1,285 @@
+
+This section describes some noteworthy details on how certain features are implemented.
+
+### \[Proposed\] FeatureX
+
+#### Proposed Implementation
+
+_{Explain here how the feature will be implemented}_
+
+Given below is an example usage scenario and how FeatureX behaves at each step.
+
+1. **Step 1**: User does something
+
+   * **Expected**: FeatureX does something in response
+
+1. **Step 2**: User does something else
+   
+      * **Expected**: FeatureX does something else in response
+
+#### Design considerations:
+
+**Aspect: How FeatureX executes:**
+
+* **Alternative 1 (current choice):** Implementation 1
+  * Pros: Easy to implement.
+  * Cons: May have performance issues in terms of memory usage.
+
+* **Alternative 2:** Implementation 2
+  * Pros: Will use less memory.
+  * Cons: We must add addtional test cases.
+
+_{more aspects and alternatives to be added}_
+
+### \[Proposed\] Data archiving
+
+_{Explain here how the data archiving feature will be implemented}_
+
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Project Guides**
+
+* [Documentation guide](Documentation.md)
+* [Testing guide](Testing.md)
+* [Logging guide](Logging.md)
+* [Configuration guide](Configuration.md)
+* [DevOps guide](DevOps.md)
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Appendix: Requirements**
+
+### Product scope
+
+**Target user profile**:
+
+* user needs
+* user preferences
+* user information
+
+**Value proposition**: Benefit compared to other similar products
+
+
+### User stories
+
+Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unlikely to have) - `*`
+
+| Priority | As a …​                                    | I want to …​                 | So that I can…​                                                        |
+|----------|--------------------------------------------|------------------------------|------------------------------------------------------------------------|
+| `* * *`  | new user                                   | see something       | visualise something                 |
+| `* *`    | user                                       | do something | complete something                |
+| `*`      | user with specific information | sort something         | find something                                                 |
+
+*{More to be added}*
+
+### Use cases
+
+(For all use cases below, the **System** is the `ProjectEx` and the **Actor** is the `User`, unless specified otherwise)
+
+**Use case: Do something**
+
+**MSS**
+
+1.  User requests ProjectEx
+2.  ProjectEx provides response
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. Requests fails.
+
+  Use case ends.
+
+* 3a. User request is invalid.
+
+    * 3a1. ProjectEx shows an error message.
+
+      Use case resumes at step 2.
+
+*{More to be added}*
+
+### Non-Functional Requirements
```

Comment:
Would move this under Product Scope unless this is part of the expected 2103T format, because a general user is more likely to list their NFRs rather than extensive user stories / use cases